tools/clang/test/HLSLFileCheck/d3dreflect/rdat_mintarget/sm6x_wavemma.hlsl

```diff
@@ -0,0 +1,53 @@
+// RUN: %dxc -T lib_6_x %s | %D3DReflect %s | %FileCheck %s -check-prefixes=RDAT
+
+// Ensure min shader target incorporates optional features used
+
+// SM 6.9+
+
+///////////////////////////////////////////////////////////////////////////////
+// ShaderFeatureInfo_WaveMMA (0x8000000) = 134217728
+
+RWByteAddressBuffer BAB : register(u1, space0);
+
+// RDAT-LABEL: UnmangledName: "use_wavematrix"
+// ShaderFeatureInfo_WaveMMA (0x8000000) = 134217728
+// + WaveOps (0x4000) = 0x8004000 = 134234112
+// RDAT:   FeatureInfo1: 134234112
+// RDAT:   FeatureInfo2: 0
+// MinShaderTarget: (Library(6) << 16) + (SM 6.9 ((6 << 4) + 9)) = 0x60069 = 393321
+// RDAT: MinShaderTarget: 393321
+
+[noinline] export
+void use_wavematrix() {
+  // Use WaveMatrix in a minimal way that survives dead code elimination.
+  WaveMatrixLeft<float, 16, 16> wml;
+  wml.Fill(0);
+  wml.Store(BAB, 0, 1024, false);
+}
+
+// RDAT-LABEL: UnmangledName: "call_use_wavematrix"
+// ShaderFeatureInfo_WaveMMA (0x8000000) = 134217728
+// + WaveOps (0x4000) = 0x8004000 = 134234112
+// RDAT:   FeatureInfo1: 134234112
+// RDAT:   FeatureInfo2: 0
+// MinShaderTarget: (Library(6) << 16) + (SM 6.9 ((6 << 4) + 9)) = 0x60069 = 393321
+// RDAT: MinShaderTarget: 393321
+
+[noinline] export
+void call_use_wavematrix() {
+  use_wavematrix();
+}
```

Comment:
I'm curious why a second level of indirection was useful here and nowhere else.

Reply:
I honestly don't recall at the moment.  I think I wanted to make sure the exported function picked up the called function as well in this case, but I don't really remember now.  It seems useful to have at least one test here that does two levels of indirection though, so I'm inclined to leave it.